cmd/libs/go2idl/generator/execute.go

```diff
@@ -0,0 +1,226 @@
+/*
+Copyright 2015 The Kubernetes Authors All rights reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package generator
+
+import (
+	"bytes"
+	"fmt"
+	"go/format"
+	"io"
+	"log"
+	"os"
+	"path/filepath"
+	"strings"
+
+	"k8s.io/kubernetes/cmd/libs/go2idl/namer"
+	"k8s.io/kubernetes/cmd/libs/go2idl/types"
+)
+
+// ExecutePackages runs the generators for every package in 'packages'. 'outDir'
+// is the base directory in which to place all the generated packages; it
+// should be a physical path on disk, not an import path. e.g.:
+// /path/to/home/path/to/gopath/src/
+// Each package has its import path already, this will be appended to 'outDir'.
+func (c *Context) ExecutePackages(outDir string, packages Packages) error {
+	for _, p := range packages {
+		if err := c.ExecutePackage(outDir, p); err != nil {
+			return err
+		}
+	}
+	return nil
+}
+
+type file struct {
+	name        string
+	packageName string
+	header      []byte
+	imports     map[string]struct{}
+	vars        bytes.Buffer
+	consts      bytes.Buffer
+	body        bytes.Buffer
+}
+
+func (f *file) assembleToFile(pathname string) error {
+	log.Printf("Assembling file %q", pathname)
+	destFile, err := os.Create(pathname)
+	if err != nil {
+		return err
+	}
+	defer destFile.Close()
+
+	b := &bytes.Buffer{}
+	et := NewErrorTracker(b)
+	f.assemble(et)
+	if et.Error() != nil {
+		return et.Error()
+	}
+	if formatted, err := format.Source(b.Bytes()); err != nil {
+		log.Printf("Warning: unable to run gofmt on %q (%v).", pathname, err)
```

Comment:
Sorry - I wasn't clear. I didn't mean removing the error - I meant removing writing to destFile in the next line.